hal/armv8m/spinlock.c

```diff
@@ -26,21 +26,21 @@ void hal_spinlockSet(spinlock_t *spinlock, spinlock_ctx_t *sc)
 {
 	/* clang-format off */
 	__asm__ volatile(" \
-		mrs r2, primask; \
+		mrs r1, primask; \
 		cpsid i; \
-		str r2, [%0]; \
-		mov r3, #0; \
+		str r1, [%0]; \
+		mov r2, #0; \
 	1: \
-		ldrexb r2, [%1]; \
-		cmp r2, #0; \
+		ldrexb r1, [%1]; \
+		cmp r1, #0; \
 		beq 1b; \
-		strexb r2, r3, [%1]; \
-		cmp r2, #0; \
+		strexb r1, r2, [%1]; \
+		cmp r1, #0; \
 		bne 1b; \
 		dmb"
 	:
 	: "r" (sc), "r" (&spinlock->lock)
-	: "r2", "r3", "memory", "cc");
+	: "r1", "r2", "memory", "cc");
```

Comment:
Why you change registers?
In your version, the compiler adds an instruction at the beginning of the function:
```
mov r3,r1
```
Same in `hal_spinlockClear()`